### PR TITLE
Improve wording for Concepts for configuring thread pools in docs

### LINKED ISSUE #26402: Improve wording in Concepts for configuring thread pools section in documentation
### Description

Some words and links to the relevant database configuration options are missing.

### Discussion

_No response_

### Motivation

_No response_

### Details

_No response_

### PR DESCRIPTION
**Pull request by thomasdarimont**
Fixes #26402

### CODE REVIEW

File: docs/guides/high-availability/concepts-threads.adoc [hunk 1]
```diff
@@ -20,18 +20,19 @@ For a configuration where this is applied, visit <@links.ha id="deploy-keycloak-
...
+When the load and the number of threads increases, the database connections can also become a bottleneck.
 Once a request cannot acquire a database connection within 5 seconds, it will fail with a message in the log like `Unable to acquire JDBC Connection`.
 The caller will receive a response with a 5xx HTTP status code indicating a server side error.
 
-If you increase the number of database connections and the number of threads too much, the system will be congested under a high load with requests queueing up and all callers getting a bad performance.
+If you increase the number of database connections and the number of threads too much, the system will be congested under a high load with requests queueing up and all callers, which leads to bad performance.
+The number of database connections is configured via the https://quarkus.io/guides/all-config#quarkus-agroal_quarkus-agroal-agroal-database-connection-pool[`Agroal - Database connection pool`] settings `quarkus.datasource.jdbc.initial-size`, `quarkus.datasource.jdbc.min-size` and `quarkus.datasource.jdbc.max-size` respectively.
```
Comment (ahus1) [inline docs/guides/high-availability/concepts-threads.adoc:35]: Th guide should refer to Keycloak's configuration options, not the Quarkus options. See https://www.keycloak.org/server/all-config#category-database for a list of Keycloak's database options.

The rationale around this: For all supported way of configuring Keycloak, there should be Keycloak configuration options. Those usually work as a facade to Quarkus and other underlying technologies. This allows us to have Keycloak specific defaults, and sometimes also hide the complexity of a underlying technolgoies.

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank you for reviewing these docs. See below for one requested change.

File: docs/guides/high-availability/concepts-threads.adoc (hunk 1 above)
Comment (thomasdarimont) [reply docs/guides/high-availability/concepts-threads.adoc]: Thanks for the review, I'll link to the keycloak database settings instead.

**Review by ahus1 — APPROVED** [review]
Thanks, LGTM

**Pull request merged**